java/ql/src/semmle/code/java/frameworks/JavaxJson.qll

```diff
@@ -0,0 +1,62 @@
+/**
+ * Provides models for the `javax.json` package.
+ */
+
+import java
+private import semmle.code.java.dataflow.ExternalFlow
+
+private class FlowSummaries extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "javax.json;JsonArray;false;getBoolean;;;Argument[-1];ReturnValue;taint",
+        "javax.json;JsonArray;false;getBoolean;;;Argument[1];ReturnValue;value",
+        "javax.json;JsonArray;false;getInt;;;Argument[-1];ReturnValue;taint",
+        "javax.json;JsonArray;false;getInt;;;Argument[1];ReturnValue;value",
+        "javax.json;JsonArray;false;getJsonArray;;;Argument[-1];ReturnValue;taint",
+        "javax.json;JsonArray;false;getJsonNumber;;;Argument[-1];ReturnValue;taint",
+        "javax.json;JsonArray;false;getJsonObject;;;Argument[-1];ReturnValue;taint",
+        "javax.json;JsonArray;false;getJsonString;;;Argument[-1];ReturnValue;taint",
+        "javax.json;JsonArray;false;getString;;;Argument[-1];ReturnValue;taint",
+        "javax.json;JsonArray;false;getString;;;Argument[1];ReturnValue;value",
+        "javax.json;JsonArray;false;getValuesAs;;;Argument[-1];ReturnValue;taint",
```

Comment:
Shouldn't that be `value` flow since [`getValueAs(Class)`](https://jakarta.ee/specifications/platform/9.1/apidocs/jakarta/json/jsonarray#getValuesAs-java.lang.Class-) seems to return a view (without modifying the entries in any way)? (or is `value` flow only when it is the exact same reference?)
Though the overload [`getValueAs(Function)`](https://jakarta.ee/specifications/platform/9.1/apidocs/jakarta/json/jsonarray#getValuesAs-java.util.function.Function-) would only be `taint` flow then I guess.

Reply:
It should be the exact same reference indeed